ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/views/MaterializedViewRewritingRelVisitor.java

```diff
@@ -127,7 +129,8 @@ public void visit(RelNode node, int ordinal, RelNode parent) {
             // If it is not a materialized view, we do not rewrite it
             throw new ReturnedValue(false);
           }
-          if (containsAggregate && !AcidUtils.isFullAcidTable(hiveTable.getHiveTableMD())) {
+          fullAcidView = AcidUtils.isFullAcidTable(hiveTable.getHiveTableMD());
```

Comment:
Is it possible to pass it through constructor? Since this class did not have static utility methods calls until now, I would try to avoid introducing it.

Reply:
It has. The change in this patch is just saves the result of the static `AcidUtils.isFullAcidTable` call to a local varialble.
Anyway, I like the idea so I moved this check outside of this class.